### PR TITLE
Add redux-snackbar messaging system.

### PR DESCRIPTION
**Pull request by plveir**
- Add  redux-snackbar messaging system,
- Apply this system to some error pattern.

### Known Bugs:
When snackbar appear first time, it causes `Warning: findDOMNode is deprecated in StrictMode. findDOMNode was passed an instance of Transition which is inside StrictMode. Instead, add a ref directly to the element you want to reference. ...`.
- this is because of Materail UI. (maybe)
- See https://github.com/mui-org/material-ui/issues/13394
- to avoid it, removing <React.StrictMode> is one of possible solutions.
- ignoring it is also one of them... 👎

### CODE REVIEW

File: src/App.js [hunk 1]
```diff
@@ -290,7 +294,11 @@ function InputDamjiroGakufu ({ dispatch }) {
             const videoId = json.youtubeVideoId
             const timeOffset = json.timeOffset
             if (!isString(videoId) || !isNumber(timeOffset))
-              throw new Error('Invalid JSON')
+              dispatch({
+                type: 'SNACK_LOAD',
+                message: 'invalid JSON',
+                variant: 'error'
+              })
```
Comment (ushitora-anqou) [inline src/App.js:301]: This call of `dispatch` won't interrupt the process, so its following lines (L.302~304) will be executed, which is not correct.
Probably `throw new Error(...)` should be untouched here, and dispatch `SNACK_LOAD` in the `catch` clause (L.306).

File: src/util/MIDILoader.js [hunk 2]
```diff
@@ -0,0 +1,18 @@
+const MIDILoader = (file, onLoad, dispatch) => {
+  try {
+    if (!['audio/midi', 'audio/x-midi', 'audio/mid'].includes(file.type)) {
+      dispatch({
+        type: 'SNACK_LOAD',
+        message: 'Invalid mime type',
+        variant: 'error'
+      })
```
Comment (ushitora-anqou) [inline src/util/MIDILoader.js:8]: Throw `Error` here, and dispatch `SNACK_LOAD` in the `catch` clause.

File: src/App.js (hunk 1 above)
Comment (plveir) [reply src/App.js:301]: >  so its following lines (L.302~304) will be executed, which is not correct.

I see, fix it. However, throwing a new error is some difficulty. See below.

File: src/util/MIDILoader.js (hunk 2 above)
Comment (plveir) [reply src/util/MIDILoader.js:8]: To fix the warning “'throw' of exception caught locally” from IDE, you should not use `throw new Error` to control flow in try-catch block. (maybe)
See https://stackoverflow.com/questions/47015693/how-to-fix-throw-of-exception-caught-locally

Comment (ushitora-anqou) [reply src/util/MIDILoader.js:8]: Hmm. We should not ignore such warnings unless we have proper reason, and I feel we don't have it here. Maybe we should leave them untouched...

**Review by ushitora-anqou — APPROVED** [review]

File: src/App.js (hunk 1 above)
Comment (plveir) [reply src/App.js:301]: Fix it to move following lines in else block.

File: src/util/MIDILoader.js (hunk 2 above)
Comment (plveir) [reply src/util/MIDILoader.js:8]: :+1:

File: src/App.js (hunk 1 above)
Comment (ushitora-anqou) [reply src/App.js:301]: Please wrap these lines in `{}`.

**Review by ushitora-anqou — APPROVED** [review]

**Pull request merged**